src/modeling_library/switch/generate.jl

```diff
@@ -0,0 +1,47 @@
+mutable struct SwitchGenerateState{T1, T2, Tr}
+    score::Float64
+    noise::Float64
+    weight::Float64
+    cond::Bool
+    subtrace::Tr
+    retval::Union{T1, T2}
+    SwitchGenerateState{T1, T2, Tr}(score::Float64, noise::Float64, weight::Float64) where {T1, T2, Tr} = new{T1, T2, Tr}(score, noise, weight)
+end
+
+function process!(gen_fn::Switch{T1, T2, Tr}, 
+                  branch_p::Float64, 
+                  args::Tuple, 
+                  choices::ChoiceMap, 
+                  state::SwitchGenerateState{T1, T2, Tr}) where {T1, T2, Tr}
+   
+    # create flip distribution
+    flip_d = bernoulli(branch_p)
```

Comment:
@femtomc This samples from a Bernoulli distribution with probability branch_p and returns a Bool.

Reply:
Apologies - that comment needs to be removed. I figured that bit out and corrected it elsewhere, but the comments indicate otherwise.